sphinx_gallery/gen_rst.py

```diff
@@ -1695,6 +1682,10 @@ def _get_callables(gallery_conf, key):
                     raise ConfigError(f"Unknown string option for {readable}: {what}")
                 what = _reset_dict[what]
             which[wi] = what
+        # make sure built-in sorter classes get instantiated (so they become callable)
+        if is_custom_sorter or what in BUILTIN_SORTERS:
```

Comment:
if we change this conditional to

```suggestion
        # make sure classes get instantiated (so they become callable)
        if key == "within_subsection_order" and inspect.isclass(what):
```

then MPL's custom sorter class should work too, and the test suite still passes locally.